src/features/posts/postsSlice.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+export const getPosts = createAsyncThunk(
+  "posts/getPosts",
+  async ({ subreddit, after = "" }) => {
+   const result = fetch(`https://www.reddit.com/r/${subreddit}.json?limit=30&after=${after}`)
+      .then((res) => res.json())
+      .then((data) => data);
+      return result
+  }
+);
+
+export const getPost = createAsyncThunk(
+  "posts/getPost",
+  async (postId) => {
+    return fetch(`https://api.reddit.com/api/info/?id=t3_${postId}`)
+      .then((res) => res.json())
+      .then((data) => Object.assign({}, data.data.children[0]));
```

Comment:
I am not sure what you are doing here, is data.data.children[0] not enough ? why Object.assign ?

Reply:
Removed Object.assign